.github/workflows/publish-docker.yml

```diff
@@ -0,0 +1,35 @@
+name: Publish to Docker
+on:
+  push:
+    branches: [ master ]
+    tags: [ 'v*' ]
+  pull_request:
+    branches: [ master ]
+ 
+env:
+  DOCKER_TAG_IMAGE: "ghcr.io/github/pages-gem"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+
+      - name: Get Docker Metadata
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: ${{ env.DOCKER_TAG_IMAGE }}
+
+      - name: Build Docker Image
+        run: |
+          docker build -t ${{ steps.meta.outputs.tags }} -f Dockerfile .
+
+      - name: Push to Container Registry
+        if: github.event_name != 'pull_request'
```

Comment:
Should we only run this job on `push` if we discard the build for the `pull_request` event?

Reply:
Discussed out of band, but I think we settled on continuing to allow this to run on PRs but only publishing to the package registry on pushes to main and tagged branches.